config/initializers/http_client_proxy.rb

```diff
@@ -11,11 +11,16 @@
     config.x.http_client_proxy[:proxy] = { proxy_address: host, proxy_port: proxy.port, proxy_username: proxy.user, proxy_password: proxy.password }.compact
     break
   end
+
+  config.x.access_to_darknet = ENV['ALLOW_ACCESS_TO_DARKNET'] == 'true'
+  config.x.darknet_via_transparent_proxy = ENV['DARKNET_VIA_TRANSPARENT_PROXY'] == 'true'
 end
 
 module Goldfinger
   def self.finger(uri, opts = {})
-    opts = opts.merge(Rails.configuration.x.http_client_proxy)
+    to_darknet = /\.(onion|i2p)(:\d+)?$/.match(uri)
```

Comment:
This doesn't seem to be specific to either I2P or Tor. Someone might want to federate with servers on Freenet, or on some mesh network or private network that requires a proxy for access.

It seems that there are two concerns: list of zones for which SSL verification is disabled, and list of zones which are accessed via a proxy.